### PR TITLE
Develop branch set with correct secrets and variables (cypress don't work with env)

### PR DESCRIPTION
**Pull request by AlexMelov**

### CODE REVIEW

File: .github/workflows/ci-dev.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 name: ci
 
-on: [ push, pull_request ]
-
+on:
+  push:
+    branches:
+      - develop
```
Comment (henryruhs) [inline .github/workflows/ci-dev.yml:6]: Empty newline after that block

File: .github/workflows/ci-dev.yml [hunk 2]
```diff
@@ -53,12 +56,13 @@ jobs:
...
             -   run: npm run start:local --prefix ./backend &
                 env:
-                    DB_URL: ${{ secrets.DB_URL }}
+                    DB_URL: ${{ secrets.DB_URL_DEV }}
             -   run: npm run start:local --prefix ./frontend &
-            -   run: npm run wait-for-serve --prefix ./frontend && npm run e2e --prefix ./frontend
+            -   run: npm run wait-for-serve --prefix ./frontend && npm run e2e:local --prefix ./frontend
                 timeout-minutes: 10
-    deploy:
+    deploy-develop:
```
Comment (henryruhs) [inline .github/workflows/ci-dev.yml]: deploy-dev

File: frontend/package.json [hunk 3]
```diff
@@ -52,7 +52,10 @@
 		"test:dev": "cross-env-shell REACT_APP_ENV=dev npm run test",
 		"test:stage": "cross-env-shell REACT_APP_ENV=stage npm run test",
 		"test:prod": "cross-env-shell REACT_APP_ENV=prod npm run test",
-		"e2e": "cypress run",
+		"e2e:local": "cross-env-shell REACT_APP_ENV=local cypress run",
```
Comment (henryruhs) [inline frontend/package.json]: Follow the pattern of existing commands...

File: frontend/src/environments/environment.local.ts [hunk 4]
```diff
@@ -11,8 +11,8 @@ export const environment : Environment =
 	apiPort: 8000,
 	apiRoutes:
 	{
-		todos: '/todos',
-		todosWithId: '/todos/:todoId'
+		todos: '/api/todos',
```
Comment (henryruhs) [inline frontend/src/environments/environment.local.ts]: Revert all /api changes

File: frontend/src/environments/environment.prod.ts [hunk 5]
```diff
@@ -7,12 +7,12 @@ export const environment : Environment =
 		branch: 'master',
 		environment: 'prod'
 	},
-	apiUrl: 'https://full-stack-a3sgu4q5h-alexmelov.vercel.app/api',
```
Comment (henryruhs) [inline frontend/src/environments/environment.prod.ts:10]: Revert all the /api changes

File: frontend/src/environments/environment.dev.ts [hunk 6]
```diff
@@ -7,12 +7,12 @@ export const environment : Environment =
 		branch: 'master',
 		environment: 'dev'
 	},
-	apiUrl: 'https://full-stack-a3sgu4q5h-alexmelov.vercel.app/api',
+	apiUrl: 'https://full-stack-a3sgu4q5h-alexmelov.vercel.app',
```
Comment (henryruhs) [inline frontend/src/environments/environment.dev.ts]: Revert the /api changes

File: backend/environments/environment.local.ts [hunk 7]
```diff
@@ -11,8 +11,8 @@ export const environment : Environment =
 	apiPort: 8000,
 	apiRoutes:
 	{
-		todos: '/todos',
-		todosWithId: '/todos/:todoId'
+		todos: '/api/todos',
```
Comment (henryruhs) [inline backend/environments/environment.local.ts]: Revert all /api changes

File: frontend/cypress/integration/todo.spec.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@ describe('todo', () =>
 {
 	beforeEach(() =>
 	{
-		cy.visit('localhost:3000');
+		cy.visit('https://todos-dev.vercel.app');
```
Comment (henryruhs) [inline frontend/cypress/integration/todo.spec.ts:5]: environment.baseUrl instead... extend all the env files

File: backend/environments/environment.local.ts (hunk 7 above)
Comment (henryruhs) [reply backend/environments/environment.local.ts]: This is not resolved... remove /api from routes

File: frontend/src/components/NewTodo.tsx [hunk 9]
```diff
@@ -12,7 +12,7 @@ const NewTodo : React.FC = () =>
 
 	async function sendTodo (title : string) : Promise<void>
 	{
-		await axios.post(environment.apiUrl + environment.apiRoutes.todos, { title });
+		await axios.post(environment.apiUrl + environment.apiRoutes.api + environment.apiRoutes.todos, { title });
```
Comment (henryruhs) [inline frontend/src/components/NewTodo.tsx]: This must be a joke... are you doing it again?

File: frontend/src/environments/environment.local.ts [hunk 10]
```diff
@@ -9,8 +9,11 @@ export const environment : Environment =
 	},
 	apiUrl: 'http://localhost',
 	apiPort: 8000,
+	baseUrl : 'http://localhost',
```
Comment (henryruhs) [inline frontend/src/environments/environment.local.ts]: I guess localhost:3000 is okay aswell

File: frontend/src/environments/environment.stage.ts [hunk 11]
```diff
@@ -9,10 +9,13 @@ export const environment : Environment =
 	},
 	apiUrl: 'https://full-stack-a3sgu4q5h-alexmelov.vercel.app',
 	apiPort: null,
+	baseUrl : 'http://localhost',
+	basePort : 3000,
 	apiRoutes:
 	{
-		todos: '/api/todos',
-		todosWithId: '/api/todos/:todoId'
+		api: '/api',
```
Comment (henryruhs) [inline frontend/src/environments/environment.stage.ts]: Remove the /api things from routes... it belongs to the apiUrl

File: frontend/src/store/todos-context.tsx [hunk 12]
```diff
@@ -33,7 +33,7 @@ const TodosContextProvider : React.FC = props =>
 
 	async function deleteRequest (_id : number, deletedItem : Todo[], nonDeletedItems : Todo[])
 	{
-		await axios.delete(environment.apiUrl + environment.apiRoutes.todos + '/' + _id, {
+		await axios.delete(environment.apiUrl + environment.apiRoutes.api + environment.apiRoutes.todos + '/' + _id, {
```
Comment (henryruhs) [inline frontend/src/store/todos-context.tsx]: I removed this 2 times now... /api belongs to end of the apiUrl

File: frontend/package.json (hunk 3 above)
Comment (henryruhs) [reply frontend/package.json]: Not resolved

**Comment by henryruhs** [comment]
PR not ready to be merged...

File: frontend/package.json (hunk 3 above)
Comment (henryruhs) [reply frontend/package.json]: "e2e:local": "cross-env-shell REACT_APP_ENV=local npm run e2e",

File: .github/workflows/ci-dev.yml [hunk 13]
```diff
@@ -1,6 +1,9 @@
 name: ci
```
Comment (henryruhs) [inline .github/workflows/ci-dev.yml]: Rename to ci-dev

File: frontend/cypress/integration/todo.spec.ts (hunk 8 above)
Comment (AlexMelov) [reply frontend/cypress/integration/todo.spec.ts:5]: Cypress is not reading the environmental files, I've changed baseUrl to hardcoded link on which cypress is working, and he is not responding

**Pull request closed without merging**